src/pet_serialization.cpp

```diff
@@ -0,0 +1,144 @@
+extern "C" {
+#include "../include/pet.h"
+}
+
+#include "../include/pet_serialization.h"
+#include "../include/vecbuf.hpp"
+
+#include <sstream>
+
+#include <boost/serialization/serialization.hpp>
+#include <boost/archive/binary_oarchive.hpp>
+#include <boost/archive/binary_iarchive.hpp>
+
+
+class BmiPetSerialization {
+    public:
+        pet_model* model;
+        BmiPetSerialization(Bmi* bmi)
+            : model{(pet_model*)bmi->data} {}
+        ~BmiPetSerialization() = default;
+        template<class Archive>
+        void serialize(Archive& ar, const unsigned int version);
+};
+
+
+template<class Archive>
+void BmiPetSerialization::
+serialize(Archive& ar, const unsigned int version) {
```

Comment:
I worry that, without extensive documentation about how this is designed to work, this function is going to be confusing for model developers who are not also highly skilled with C++ (and maybe also pretty familiar with Boost).   And complete lack of any documentation seems unacceptable.